components/layout/construct.rs

```diff
@@ -629,8 +630,12 @@ impl<'a> FlowConstructor<'a> {
     /// to happen.
     fn build_flow_for_nonfloated_block(&mut self, node: &ThreadSafeLayoutNode)
                                        -> ConstructionResult {
```

Comment:
How about creating a function that abstracts over the creation of both floated and nonfloated block flows (including multicol) then, and redirecting all code that creates block flows to use it? I'm wary of bugs from having to have multiple code paths that all have to check to see whether they need to create a multicolumn block, since we've had similar bugs relating to tables and list item flows in the past.